### PR TITLE
Legend and Data Dict tabs in generated templates via --legend flag

### PR DESCRIPTION
**Pull request by curlup**

### CODE REVIEW

File: cidc_schemas/template_writer.py [hunk 1]
```diff
@@ -98,13 +98,15 @@ def __init__(
         data_rows=_DATA_ROWS,
         min_num_cols=_MIN_NUM_COLS,
         column_width_px=_COLUMN_WIDTH_PX,
+        with_legend_and_dd=False,
```
Comment (jacoblurye) [inline cidc_schemas/template_writer.py]: Could this be an argument to `XlTemplateWriter.write` instead?

File: cidc_schemas/template_writer.py [hunk 2]
```diff
@@ -124,9 +126,139 @@ def write(self, outfile_path: str, template: Template):
...
             first_sheet = False
 
+        if self.with_legend_and_dd:
+            self._write_legend(self.template.worksheets)
+            self._write_data_dict(self.template.worksheets)
+
         self.workbook.close()
         self.workbook = None
 
+    def _write_data_dict(self, schemas):
```
Comment (jacoblurye) [inline cidc_schemas/template_writer.py]: A comment about what "data dictionary" actually means here (or somewhere else) would be helpful. It's not obvious to someone without outside context that this is a list of allowed values for `enum` fields.

File: cidc_schemas/template_writer.py [hunk 3]
```diff
@@ -124,9 +126,139 @@ def write(self, outfile_path: str, template: Template):
...
+
+        if not len(prop_schema.get("enum")):
+            raise Exception(f"Enum {name} with no options detected:\n{prop_schema}")
+
+        for i, enum_option in enumerate(prop_schema.get("enum")):
+            ws.write(1 + i, col_n, enum_option)
+
+        return True
+
+    def _write_legend(self, schemas):
```
Comment (jacoblurye) [inline cidc_schemas/template_writer.py]: Same goes for what "legend" means.

**Comment by curlup** [comment]
> When do we expect this to be used? Do we anticipate that anyone external will use the cidc_schemas CLI?

for now only to communicate with NCI, but we might later add those tabs to all the templates.

File: cidc_schemas/template_writer.py (hunk 1 above)
Comment (curlup) [reply cidc_schemas/template_writer.py]: right, thanks!

**Pull request merged**